app/models/game.rb

```diff
@@ -2,9 +2,20 @@ class Game < ActiveRecord::Base
   belongs_to :white_player, class_name: 'User'
   belongs_to :black_player, class_name: 'User'
   belongs_to :winner, class_name: 'User'
-  has_many :pieces
+  has_many :pieces, dependent: :destroy
 
-  def open_seat?
-    white_player_id.nil? || black_player_id.nil?
+  after_create :open_seat
+
+  def open_seat
+    white_player_open = white_player_id.nil?
+    black_player_open = black_player_id.nil?
+
+    self.available = if white_player_open || black_player_open
```

Comment:
Im pretty sure this line could be slimmed down to just

self.available = white_player_open || black_player_open

It just makes it look a little cleaner.